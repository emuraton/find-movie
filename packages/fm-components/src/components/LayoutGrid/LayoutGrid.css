.LayoutGrid {
  display: flex;
  flex-wrap: wrap;
  margin: -1em 0 0 -1em;
}

.LayoutGrid-cell {
  flex: 1;
  padding: 1em 0 0 1em;

}
