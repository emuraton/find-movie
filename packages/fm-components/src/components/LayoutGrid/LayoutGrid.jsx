import React, { Component } from 'react'
import PropTypes from 'prop-types'
import env from '../../../env.json'

require('./LayoutGrid.css')

export default class LayoutGrid extends Component {
  constructor(props) {
    super(props)
    this.grid = null
  }

  getChildElements = () => {
    const { data } = this.props
    return data.results.map((element) => {
      return (
        element.poster_path ?
          <div className="LayoutGrid-cell" key={element.id}>
            <a href="#">
              <img src={`${env.IMG_URL}/w300${element.poster_path}`} alt="Movie" />
            </a>
          </div> :
        null
      )
    })
  }

  render() {
    const { data } = this.props
    if (data.length === 0 || data.results === 0) return null
    return (
      <div className="LayoutGrid">
        {this.getChildElements()}
      </div>
    )
  }
}

LayoutGrid.PropTypes = {
  data: PropTypes.array.isRequired
}

LayoutGrid.defaultProps = {
  data: []
}
