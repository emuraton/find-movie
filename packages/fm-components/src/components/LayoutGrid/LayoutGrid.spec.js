import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import LayoutGrid from './LayoutGrid'

describe('<LayoutGrid />', () => {

  const initialProps = {
    data: {
      results:
      [
        {
          id: '1',
          poster_path: 'myPath1'
        },
        {
          id: '2',
          poster_path: 'myPath2'
        },
        {
          id: '3',
          poster_path: 'myPath3'
        }
      ]
    }
  }

  describe('@renders', () => {
    it('in default state', () => {
      const wrapper = shallow(<LayoutGrid {...initialProps} />)
      expect(toJson(wrapper))
        .toMatchSnapshot()
    })

    it('with empty data', () => {
      const props = {
        ...initialProps,
        data: []
      }
      const wrapper = shallow(<LayoutGrid {...initialProps} />)
      expect(toJson(wrapper))
        .toMatchSnapshot()
    })
  })
})
