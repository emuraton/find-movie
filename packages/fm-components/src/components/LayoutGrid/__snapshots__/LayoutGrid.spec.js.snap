// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LayoutGrid /> @renders in default state 1`] = `
<div
  className="LayoutGrid"
>
  <div
    className="LayoutGrid-cell"
  >
    <a
      href="#"
    >
      <img
        src="https://image.tmdb.org/t/p/w300myPath1"
      />
    </a>
    <span>
      TEST
    </span>
  </div>
  <div
    className="LayoutGrid-cell"
  >
    <a
      href="#"
    >
      <img
        src="https://image.tmdb.org/t/p/w300myPath2"
      />
    </a>
    <span>
      TEST
    </span>
  </div>
  <div
    className="LayoutGrid-cell"
  >
    <a
      href="#"
    >
      <img
        src="https://image.tmdb.org/t/p/w300myPath3"
      />
    </a>
    <span>
      TEST
    </span>
  </div>
</div>
`;

exports[`<LayoutGrid /> @renders with empty data 1`] = `
<div
  className="LayoutGrid"
>
  <div
    className="LayoutGrid-cell"
  >
    <a
      href="#"
    >
      <img
        src="https://image.tmdb.org/t/p/w300myPath1"
      />
    </a>
    <span>
      TEST
    </span>
  </div>
  <div
    className="LayoutGrid-cell"
  >
    <a
      href="#"
    >
      <img
        src="https://image.tmdb.org/t/p/w300myPath2"
      />
    </a>
    <span>
      TEST
    </span>
  </div>
  <div
    className="LayoutGrid-cell"
  >
    <a
      href="#"
    >
      <img
        src="https://image.tmdb.org/t/p/w300myPath3"
      />
    </a>
    <span>
      TEST
    </span>
  </div>
</div>
`;
